asmcomp/selectgen.ml

```diff
@@ -42,6 +42,10 @@ let oper_result_type = function
   | Cintoffloat -> typ_int
   | Craise _ -> typ_void
   | Ccheckbound _ -> typ_void
+  | Cperform -> typ_addr
```

Comment:
It looks to me like having a parameterized result type like Capply might allow slightly better code generation when the result of these operations is not used because it can be turned into `typ_void`, but I'm not very familiar with this part of the compiler so I could be wrong.

I'm also surprised that the results are `typ_addr` rather than `typ_val` since I would have thought they returned valid OCaml values that should be made GC roots.